### PR TITLE
Fix typo in CI

### PR DESCRIPTION
**Pull request by szepeviktor**
Although there were only 2 typos: `$({` I've used a function to concatenate the two parts.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -200,12 +200,12 @@ jobs:
           key: phpstan-${{ github.ref_name }}
 
       - name: Clear previous version of PHPStan main cache
-        if: github.ref == 'refs/heads/$({ env.MAIN_BRANCH }}'
+        if: ${{ github.ref == format('refs/heads/{0}', env.MAIN_BRANCH) }}
```
Comment (szepeviktor) [inline .github/workflows/ci.yml:203]:
```suggestion
        if: ${{ github.ref_name == github.event.repository.default_branch }}
```
What do you think about it?

Comment (PrinsFrank) [reply .github/workflows/ci.yml:203]: That would be great, didn't know this was possible!

Comment (szepeviktor) [reply .github/workflows/ci.yml]: Done.

**Comment by szepeviktor** [comment]
I've developed `github` context dumping for you: https://github.com/szepeviktor/byte-level-care/actions/runs/7449368212/job/20265848794

**Comment by PrinsFrank** [comment]
@szepeviktor Haha, thanks!

**Pull request merged**